crates/iota/src/client_commands.rs

```diff
@@ -1526,15 +1526,41 @@ impl IotaClientCommands {
                 match (amounts.as_ref(), count) {
                     (None, None) => bail!("You must use one of amounts or count options."),
                     (Some(_), Some(_)) => bail!("Cannot specify both amounts and count."),
-                    (None, Some(0)) => bail!("Coin split count must be greater than 0"),
+                    (None, Some(0)) | (None, Some(1)) => {
+                        bail!("Coin split count must be greater than 1")
+                    }
                     _ => { /*no_op*/ }
                 }
                 let client = context.get_client().await?;
-                let tx_kind = client
-                    .transaction_builder()
-                    .split_coin_tx_kind(coin_id, amounts, count)
-                    .await?;
                 let signer = context.get_object_owner(&coin_id).await?;
+                let gas_coins_page = client
+                    .coin_read_api()
+                    .get_coins(signer, None, None, None)
+                    .await?;
+                // If we only have a single coin, we have to split from the gas coin
+                let tx_kind = if gas_coins_page.data.len() == 1 {
+                    if let Some(amounts) = amounts {
+                        client
+                            .transaction_builder()
+                            .pay_iota_tx_kind(vec![signer; amounts.len()], amounts)?
```

Comment:
Can we not use the actual split command?

Reply:
We can't change the internal functionality?

Reply:
As far as I know not, as we don't know if the provided object id is also used as gas coin or not

Reply:
Then where is it failing? I'm confused.

Reply:
At this point the transaction is built assuming the provided coin is not the gas object, the gas object is selected later. If the gas object is the same as the provided object it will fail, as it was provided twice as input. Without changing the params to the functions we can't change the assumption, that this object is not used also as gas object

Reply:
Yes, but why not change it where it checks for the gas object so that this doesn't fail when you try to call it this way. Otherwise using this method via the SDKs will still fail

Reply:
The gas object is selected after this step, you can't go back and change it then (at least not without super complicated logic to decode it again and replace stuff)